packages/kubernetes-1.20/kubelet-config

```diff
@@ -29,7 +29,11 @@ authorization:
 clusterDomain: {{settings.kubernetes.cluster-domain}}
 {{#if settings.kubernetes.cluster-dns-ip}}
 clusterDNS:
+{{#each settings.kubernetes.cluster-dns-ip}}
+- {{this}}
+{{else}}
 - {{settings.kubernetes.cluster-dns-ip}}
+{{/each}}
```

Comment:
I'm a bit surprised that this works - `#each` iterates over a list and `else` triggers if it's defined but not actually a list?

Reply:
I was pleased to discover this behavior, because otherwise we would need a custom helper which would get a bit messy. But yes, I looked [through the handlebars source](https://github.com/sunng87/handlebars-rust/blob/master/src/helpers/helper_each.rs#L143) and noticed that `#each` will render the `inverse()` statement in the template (which is the `else`) if the given value is not a list or object.

This is [consistent with the javascript implementation](https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23each%20listInput%7D%7D%5Cnlist%20%7B%7Bthis%7D%7D%5Cn%7B%7Belse%7D%7D%5Cnother%20%7B%7BlistInput%7D%7D%5Cn%7B%7B%2Feach%7D%7D%5Cn%5Cn%7B%7B%23each%20strInput%7D%7D%5Cnlist%20%7B%7Bthis%7D%7D%5Cn%7B%7Belse%7D%7D%5Cnother%20%7B%7BstrInput%7D%7D%5Cn%7B%7B%2Feach%7D%7D%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20listInput%3A%20%5B%5C%22test1%5C%22%2C%20%5C%22test2%5C%22%5D%2C%5Cn%20%20strInput%3A%20%5C%22test3%5C%22%2C%5Cn%7D%5Cn%22%2C%22output%22%3A%22list%20test1%5Cnlist%20test2%5Cn%5Cnother%20test3%5Cn%22%2C%22preparationScript%22%3A%22%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D) as well.